pkg/executor/point_get.go

```diff
@@ -203,15 +180,19 @@ func matchPartitionNames(pid int64, partitionNames []ast.CIStr, pi *model.Partit
 }
 
 // Recreated based on Init, change baseExecutor fields also
-func (e *PointGetExecutor) Recreated(p *plannercore.PointGetPlan) {
-	e.Init(p)
+func (e *PointGetExecutor) Recreated(p *plannercore.PointGetPlan, ctx sessionctx.Context) {
 	// It's necessary to at least reset the `runtimeStats` of the `BaseExecutor`.
 	// As the `StmtCtx` may have changed, a new index usage reporter should also be created.
-	e.BaseExecutor = exec.NewBaseExecutor(e.Ctx(), p.Schema(), p.ID())
-	e.indexUsageReporter = buildIndexUsageReporter(e.Ctx(), p, false)
+	e.BaseExecutor = exec.NewBaseExecutor(ctx, p.Schema(), p.ID())
+	e.indexUsageReporter = buildIndexUsageReporter(ctx, p, false)
+
+	e.Init(p)
+	InitSnapshotWithSessCtx(e.snapshot, ctx, nil)
```

Comment:
The oreder here is different from `buildPointGet` in which the `getSnapshot` is called before `Init`, need to ensure there is no implicit order requirement bettween `Init` and getSnapshot`.